packages/plugin-commands-import/src/import.ts

```diff
@@ -11,33 +12,56 @@ import { install, InstallOptions } from 'supi'
 import rimraf from '@zkochan/rimraf'
 import loadJsonFile from 'load-json-file'
 import renderHelp from 'render-help'
+import { parse as parseYarnLock } from '@yarnpkg/lockfile'
 
 export const rcOptionsTypes = cliOptionsTypes
 
 export function cliOptionsTypes () {
-  return {}
+  return {
+    yarn: Boolean,
+  }
 }
 
 export function help () {
   return renderHelp({
-    description: `Generates ${WANTED_LOCKFILE} from an npm package-lock.json (or npm-shrinkwrap.json) file.`,
+    description: `Generates ${WANTED_LOCKFILE} from an npm package-lock.json (or npm-shrinkwrap.json, yarn.lock) file.`,
     url: docsUrl('import'),
-    usages: ['pnpm import'],
+    descriptionLists: [
+      {
+        title: 'Options',
+        list: [
+          {
+            description: `Generate ${WANTED_LOCKFILE} from yarn.lock file`,
+            name: '--yarn',
+          },
+        ],
+      },
+    ],
+    usages: [
+      'pnpm import',
+      'pnpm import --yarn',
+    ],
   })
 }
 
 export const commandNames = ['import']
 
 export async function handler (
-  opts: CreateStoreControllerOptions & Omit<InstallOptions, 'storeController' | 'lockfileOnly' | 'preferredVersions'>
+  opts: CreateStoreControllerOptions & Omit<InstallOptions, 'storeController' | 'lockfileOnly' | 'preferredVersions'> & Partial<Pick<Config, 'cliOptions'>>
 ) {
   // Removing existing pnpm lockfile
   // it should not influence the new one
   await rimraf(path.join(opts.dir, WANTED_LOCKFILE))
-  const npmPackageLock = await readNpmLockfile(opts.dir)
   const versionsByPackageNames = {}
-  getAllVersionsByPackageNames(npmPackageLock, versionsByPackageNames)
-  const preferredVersions = getPreferredVersions(versionsByPackageNames)
+  let preferredVersions = {}
+  if (opts?.cliOptions && opts.cliOptions['yarn']) {
```

Comment:
why is this option needed? We could just check if a `yarn.lock` file exists. If it exists, import it, if not, import the npm lockfile.

Reply:
In Some project, the package-lock.json and yarn.lock file both exist,i just want to make a separate of them.

Reply:
It sounds like an edge case. In those rare cases the user may rename or remove one of the lockfiles before running `pnpm import`

Reply:
Maybe i think it's normal in the development，Some people even don't konw what the lock file was doing...So i think this option maybe necessary...

Reply:
I don't get it. How is having two lockfiles at the same time is normal? You either use Yarn or you use npm, combining two different package managers is a really bad idea.

Reply:
I mean that if a project was developed by many people...

Reply:
I think it's much better to add a option here, maybe next time Some new option will be added，and users can understand the usage of command more clear.🤔

Reply:
emm, maybe you are right, i will adjust the logic here~